src/core/providers/helpers/ledger/createLedgerProvider.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  IProvider,
+  ProviderTypeEnum
+} from 'core/providers/types/providerFactory.types';
+import { getLedgerProvider } from './getLedgerProvider';
+import { fetchAccount } from 'utils/account/fetchAccount';
+import { setLedgerLogin } from 'store/actions/loginInfo/loginInfoActions';
+import { setLedgerAccount } from 'store/actions/account/accountActions';
+
+export const createLedgerProvider = async () => {
+  const data = await getLedgerProvider();
+
+  if (!data) {
+    return null;
+  }
+
+  const { ledgerProvider: provider, ledgerConfig } = data;
+
+  const createdProvider = provider as unknown as IProvider;
+
+  const hwProviderLogin = provider.login;
+
+  createdProvider.getType = () => {
+    return ProviderTypeEnum.ledger;
```

Comment:
This could be an implicit return, but can stay like it is if you think it'll be needed to debug later.